numpy/f2py/crackfortran.py

```diff
@@ -1537,10 +1552,10 @@ def markinnerspaces(line):
             continue
         if f == 0 and c in ['\'', '"']:
             cc = c
-        if c == cc:
-            f = f + 1
-        elif c == cc:
-            f = f - 1
+        if c == cc and f == 0:
+            f = 1
+        elif c == cc and f == 1:
+            f = 0
```

Comment:
Now that there's no arithmetic on `f`, can we make it a bool?

Reply:
done, also tried to give meaningful names to variables

Reply:
resolving, it seems my change passes tests.